### PR TITLE
Refactor some uses of randomIndexCompatibleVersion to use IndexVersion

### PR DESCRIPTION
**Pull request by thecoop**
Refactor most remaining uses of `VersionUtils.randomIndexCompatibleVersion` to `IndexVersion`

### CODE REVIEW

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ClientHelperTests.java [hunk 1]
```diff
@@ -391,7 +389,6 @@ public void testGetPersistableSafeSecurityHeaders() throws IOException {
         final ClusterState clusterState = mock(ClusterState.class);
         final DiscoveryNodes discoveryNodes = mock(DiscoveryNodes.class);
         when(clusterState.nodes()).thenReturn(discoveryNodes);
-        when(discoveryNodes.getMinNodeVersion()).thenReturn(VersionUtils.randomPreviousCompatibleVersion(random(), Version.CURRENT));
```
Comment (stu-elastic) [inline x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ClientHelperTests.java:394]: Can you explain this?

[the remark this answers is not in the document]
Comment (thecoop) [reply x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ClientHelperTests.java:394]: See https://github.com/elastic/elasticsearch/commit/75ac43665907a53660abe113d99e02b18cb09487 - the call was changed from `getMinNodeVersion` to `getMinTransportVersion`.

**Review by stu-elastic — APPROVED** [review]
lgtm

File: server/src/test/java/org/elasticsearch/action/search/SearchQueryThenFetchAsyncActionTests.java [hunk 2]
```diff
@@ -236,7 +237,11 @@ public void run() {
 
     public void testMinimumVersionSameAsNewVersion() throws Exception {
         Version newVersion = Version.CURRENT;
-        Version oldVersion = VersionUtils.randomPreviousCompatibleVersion(random(), newVersion);
+        Version oldVersion = VersionUtils.randomVersionBetween(
```
Comment (thecoop) [inline server/src/test/java/org/elasticsearch/action/search/SearchQueryThenFetchAsyncActionTests.java:240]: This changes from min index compatibility to min node compatibility

**Comment by thecoop** [comment]
@elasticmachine update branch

**Comment by thecoop** [comment]
@elasticmachine update branch

**Pull request merged**